### PR TITLE
[WIP]Buffs food, nerfs food

### PR DESCRIPTION
**Pull request by Very-Soft**
Once upon a time, someone thought it was a good idea to make most food just give 1/3rd as much nutrition, because protein shakes were too good, which made most food that wasn't protein shakes suck

I un-nerfed food, and just nerfed protein shakes so they give about what people expect them to now. I also undid what appears to have been an attempt at buffing food that doubled nutriment amount when it is set on the var, as with the un-nerfed nutrition conversion, that was making common food give bookoo nutrition.

At their default values, a lot of foods are much more viable now in comparison to protein shakes. Protein shakes are still one of the best vendor foods you can get, but 2 bags of jerky gives about the same amount of nutrition as 1 container of protein shake, for reference.

On top of this, I added a bunch more positive traits for food preference! These new traits are in the positive field because they give a big boost to nutrition gained when ingesting certain kinds of allergens. Herbivores for example gain extra nutrition by eating vegitables, and fruits. Carnivores gain extra nutrition by eating meat, and Omnivores gain extra nutrition from eating meat or vegitables. These traits all have 0 cost, but you can only pick one at a time, because they're pretty powerful. The hope is that when using one of these traits, you can make your character's favorite kind of food be a better choice than something like a protein shake. You can even gain nutrition from drinking coffee now!

Ultimately, I just think it would be cool if there were more different kinds of food that is actually worth seeking out, and I think it would be extra cool if, what that food IS, is different from person to person!

### CODE REVIEW

**Review by Novacat — APPROVED** [review]

**Comment by Heroman3003** [comment]
Seems fine so far

File: code/modules/mob/living/carbon/human/species/station/traits_vr/positive.dm [hunk 1]
```diff
@@ -269,3 +269,229 @@
...
+	/datum/trait/positive/food_pref/glycovore,
+	/datum/trait/positive/food_pref/lactovore,
+	/datum/trait/positive/food_pref/coffee,
+	/datum/trait/positive/food_pref/stimulant
+	)
+	our_allergens = list(ALLERGEN_BEANS)
+
+/datum/trait/positive/food_pref/omnivore
+	name = "Food Preference - Omnivore"
+	desc = "You prefer to eat meat and vegitables, and gain extra nutrition for doing so!"
```
Comment (TheToaster98) [inline code/modules/mob/living/carbon/human/species/station/traits_vr/positive.dm:339]: vegitables -> vegetables

File: code/modules/mob/living/carbon/human/species/station/traits_vr/positive.dm [hunk 2]
```diff
@@ -269,3 +269,229 @@
...
+	var/list/our_allergens = list(ALLERGEN_MEAT)
+
+/datum/trait/positive/food_pref/apply(datum/species/S, mob/living/carbon/human/H, trait_prefs)
+	. = ..()
+	for(var/a in our_allergens)
+		S.food_preference |= a
+
+/datum/trait/positive/food_pref/herbivore
+	name = "Food Preference - Herbivore"
+	desc = "You prefer to eat fruits and vegitables, and gain extra nutrition for doing so!"
```
Comment (TheToaster98) [inline code/modules/mob/living/carbon/human/species/station/traits_vr/positive.dm:303]: vegitables -> vegetables

File: code/modules/mob/living/carbon/human/species/station/traits_vr/positive.dm (hunk 1 above)
Comment (TheToaster98) [inline code/modules/mob/living/carbon/human/species/station/traits_vr/positive.dm:339]: vegitables -> vegetables

File: code/modules/mob/living/carbon/human/species/station/traits_vr/positive.dm [hunk 3]
```diff
@@ -269,3 +269,229 @@
...
+	/datum/trait/positive/food_pref/cocoavore,
+	/datum/trait/positive/food_pref/glycovore,
+	/datum/trait/positive/food_pref/lactovore,
+	/datum/trait/positive/food_pref/stimulant
+	)
+	our_allergens = list(ALLERGEN_COFFEE)
+
+/datum/trait/positive/food_pref/stimulant
+	name = "Food Preference - Stimulant Dependant"
+	desc = "You can get by on caffine alone if you have to, and you like it that way."
```
Comment (TheToaster98) [inline code/modules/mob/living/carbon/human/species/station/traits_vr/positive.dm:483]: caffine -> caffeine

**Review by TheToaster98 — COMMENTED** [review]
Found the following typos

**Pull request closed without merging**